client/starwhale/api/_impl/job.py

```diff
@@ -0,0 +1,208 @@
+        self.jobs = {}
+
+
+# shared memory, not thread safe
+# parse_config = {"parse_stage": False, "jobs": {}}
+parse_config = ParseConfig(False, {})
+
+
+class Parser:
+    @staticmethod
+    def set_parse_stage(parse_stage: bool) -> None:
+        parse_config.parse_stage = parse_stage
+
+    @staticmethod
+    def is_parse_stage() -> bool:
+        return parse_config.parse_stage
+
+    @staticmethod
+    def add_job(job_name: str, step: Step) -> None:
+        _jobs = parse_config.jobs
+        if job_name not in _jobs:
+            parse_config.jobs[job_name] = []
+
+        parse_config.jobs[job_name].append(step)
+
+    @staticmethod
+    def get_jobs() -> t.Dict[str, t.List[Step]]:
+        return parse_config.jobs
+
+    # load is unique,so don't need to think multi load and clean
+    @staticmethod
+    def clear_config() -> None:
+        global parse_config
+        parse_config.clear()
+
+    @staticmethod
+    def parse_job_from_module(module: str, path: Path) -> t.Dict[str, t.List[Step]]:
+        """
+        parse @step from module
+        :param module: module name
+        :param path: abs path
+        :return: jobs
+        """
+        Parser.set_parse_stage(True)
+        # parse DAG
+        logger.debug("parse @step for module:{}", module)
+        load_module(module, path)
+        _jobs = Parser.get_jobs().copy()
+        Parser.clear_config()
+        return _jobs
+
+    @staticmethod
+    def generate_job_yaml(module: str, path: Path, target_file: Path) -> None:
+        """
+        generate job yaml
+        :param target_file: yaml target path
+        :param module: module name
+        :param path: abs path
+        :return: None
+        """
+        _jobs = Parser.parse_job_from_module(module, path)
+        # generate DAG
+        logger.debug("generate DAG")
+        if Parser.check(_jobs):
+            # dump to target
+            # ensure_file(target_file, yaml.safe_dump(_jobs, default_flow_style=False))
+            with open(target_file, "w") as file:
+                yaml.dump(_jobs, file)
+            logger.debug("generator DAG success!")
+        else:
+            logger.error("generator DAG error! reason:{}", "check is failed.")
+
+    @staticmethod
+    def check(jobs: t.Dict[str, t.List[Step]]) -> bool:
+        # check
+        checks = []
+        logger.debug(f"jobs:{jobs}")
+        for job in jobs.items():
+            all_steps = []
+            needs = []
+            for _step in job[1]:
+                all_steps.append(_step.step_name)
+                for d in _step.needs:
+                    if d:
+                        needs.append(d)
+            logger.debug("all steps:{}, length:{}", all_steps, len(all_steps))
+            _check = all(item in all_steps for item in needs)
+            if not _check:
+                logger.error("job:{} check error!", job[0])
+            checks.append(_check)
+        # all is ok
+        if all(c is True for c in checks):
```

Comment:
`if all(c is True for c in checks):` -> `if all(checks):` ?

Reply:
```
if all(c is True for c in checks):
    logger.debug("check success! \n{}", yaml.dump(jobs))
    return True
else:
     return False
```
-->
```
return all(checks)
```

Reply:
done